### ISSUE TITLE
User not able to see voicemail box v4.2 (and 4.2.1)

### ISSUE THREAD

**Issue by andyfor** [fusionpbx/fusionpbx#2253]
Using Fusion v4.2 (and presumably 4.2.1) and Freeswitch 1.6.12 I am having issues with group permissions for users to see their voicemail box.

- Create a standard user and an extension tied to that user with voicemail enabled
- Ensure menu manager has permissions to see Apps>voicemails
- Ensure group permissions for the user in question (not admin) for voicemails has voicemail_view and voicemail_message_view enabled 
- Log in as user and go to Apps > Voicemails. Cannot see voicemail boxes assigned to user and therefore voicemail messages are hidden.
- Update group permissions for user such that voicemail_delete is checked and logout and login and can now see voicemails. As per https://github.com/fusionpbx/fusionpbx/blob/4.2/app/voicemails/resources/classes/voicemail.php line 117 it seems this has an over-ride capability but it is masking the true issue perhaps.

Thanks
Andy

**Comment by jebsolutions** [comment]
@andyfor 

I'm seeing the same thing on the master branch.

As a work around, try clicking on the voicemail link that shows up on the dashboard.  This link goes directly into listing the voicemails in your mailbox.

**Comment by andyfor** [comment]
Hi

Thanks for responding jebsolutions and good to know it's on the master branch too - very surprised nobody else has brought this on up as it seems like an obvious one.

It's strange but i don't see it on the dashboard when i log in as a user (shows dashboard but nothing under it). Yet it's enabled in the menu permissions. Is there something else that needs to be enabled perhaps?

Thanks a lot for providing feedback.
Andy

**Comment by markjcrane** [comment]
I just tested on my install and it worked fine via the dashboard. The voicemail for the regular user is meant to be accessed from the dashboard. A user by default after login will see voicemail, missed calls and recent calls. 

In advanced -> default settings there is a category called 'dashboard' it is very likely that you don't have all the defaults for that category. If you update to latest stable or master then run upgrade -> app defaults then it will likely add more settings to your dashboard category if it does not then delete all setting in that category and run upgrade -> app defaults.

Apps -> Voicemail is linked to /app/voicemails/voicemails.php it was designed for users in the admin and superadmin groups that is where you are sending your regular user and it wasn't designed to be used this way.

The user dashboard voicemail section links to /app/voicemails/voicemail_messages.php. 
If you want to you can setup a new item in the menu for users  App -> Voicemails that is linked to   /app/voicemails/voicemail_messages.php. I think I will add this by default it may help to prevent this confusion in the future.

**Comment by andyfor** [comment]
Thanks Mark. Quick note, just for completeness in case others are in the same boat.

I have some custom group permissions setup on my system so I needed to add them to Advanced > Default Settings > Dashboard in order for voicemails etc to appear in the dashboard. i.e. copy user settings to my custom group essentially.

This fixed the issue and I removed Apps > Voicemails option based on your comments above, since that functionality is not intended for basic portal access. Adding a link to voicemail_messages would make sense though I feel.

Thanks again.
Andy
